[core-v2.3.0-alpha.1] : Feature - Transaction notifications (#1070)
blocknative/web3-onboard: packages/core/README.md

```

**`notify`**
An object that defines whether transaction notifications will display (defaults to true if an API key is provided). This object contains an `enabled` flag prop and an optional `transactionHandler` which is a callback that can disable or allow customizations of notifications. 
Currently notifications positions in the same location as the account center(either below - if the Account Center is positioned along the top, or above if positioned on the bottom of the view).
The transactionHandler can react off any property of the Ethereum TransactionData returned to the callback from the event(see console.log in example init) and in turn can return a Custom Notification object to define the verbiage, styling or add functionality:
 - `Notification.message` - to completely customize the message shown 
 - `Notification.eventCode` - Handle codes in your own way - see codes here under the notify prop [default en file here](src/i18n/en.json)
 - `Notification.type` - icon type displayed (see NotificationType below for options)
 - `Notification.autoDismiss` - time on screen 
 - `Notification.link` - add link to hash 
 - `Notification.onClick()` - notification body click event

>  Notify can also be completely styled by using the CSS variables found below. These are setup to allow for maximum customization with base styling variables setting to global the global more themed styling (i.e. `--onboard-grey-600`) as well as a more precise component level styling available (`--notify-onboard-grey-600`) with the latter taking precedent if defined

Author: That sentence makes 0 sense, lol thanks for pointing that out!

  eventCode: string
  message: string
  autoDismiss: number
  link?: string
  onclick?: (event: Event) => void
}

export type NotificationType = 'pending' | 'success' | 'error' | 'hint'

export declare type Network = 'main' | 'testnet' | 'ropsten' | 'rinkeby' | 'goerli' | 'kovan' | 'xdai' | 'bsc-main' | 'matic-main' | 'fantom-main' | 'matic-mumbai' | 'local';

export interface UpdateNotification {
  (notificationObject: CustomNotification): {
    dismiss: () => void
    update: UpdateNotification
  }
}
```

 Notify can be used to deliver custom DApp notifications by passing a `CustomNotification` object to the `customNotification` action and will be returned a `UpdateNotification` type. 
 This `UpdateNotification` will return an `update` function that can be passed a new `CustomNotification` to update the existing notification.
 The `customNotification` method also returns a `dismiss` method that is called without any parameters to dismiss the notification. 
 
```typescript
  const { update, dismiss } =
    onboard.state.actions.customNotification({
      type: 'pending',
      message:
        'This is a custom DApp pending notification to use however you want',

>       autoDismiss: 0

Author: autoDismiss of `0` will keep the notification on screen until the user dismisses

>       autoDismiss: 0

Author: Added this to the documentation as well, good catch!

The `transactionHandler` can react off any property of the Ethereum TransactionData returned to the callback from the event (see console.log in example init). In turn, it can return a Custom `Notification` object to define the verbiage, styling, or add functionality:
 - `Notification.message` - to completely customize the message shown 
 - `Notification.eventCode` - handle codes in your own way - see codes here under the notify prop [default en file here](src/i18n/en.json)
 - `Notification.type` - icon type displayed (see `NotificationType` below for options)
 - `Notification.autoDismiss` - time (in ms) after which the notification will be dismissed. If set to `0` the notification will remain on screen until the user dismisses the notification, refreshes the page or navigates away from the site with the notifications 
 - `Notification.link` - add link to the transaction hash. For instance, a link to the transaction on etherscan
 - `Notification.onClick()` - onClick handler for when user clicks the notification element

>  Notify can also be completely styled by using the CSS variables found below. These are setup to allow for maximum customization with base styling variables setting to global the global more themed styling (i.e. `--onboard-grey-600`) as well as a more precise component level styling available (`--notify-onboard-grey-600`) with the latter taking precedent if defined

Author: ```suggestion
 Notify can also be styled by using the CSS variables found below. These are setup to allow maximum customization with base styling variables setting the global theme (i.e. `--onboard-grey-600`) along with more precise component level styling variables available (`--notify-onboard-grey-600`) with the latter taking precedent if defined
```
